vyper/old_codegen/expr.py

```diff
@@ -375,25 +375,7 @@ def parse_Attribute(self):
             if key == "msg.sender" and not self.context.is_internal:
                 return LLLnode.from_list(["caller"], typ="address", pos=getpos(self.expr))
             elif key == "msg.data" and not self.context.is_internal:
-                is_len = self.expr._metadata.get("is_len")
-                if is_len is True:
-                    typ = ByteArrayType(32)
-                    pos = self.context.new_internal_variable(typ)
-                    node = ["seq", ["mstore", pos, "calldatasize"], pos]
-                    return LLLnode.from_list(
-                        node, typ=typ, pos=getpos(self.expr), location="memory"
-                    )
-                size = self.expr._metadata.get("size")
-                typ = ByteArrayType(size + 32)
-                pos = self.context.new_internal_variable(typ)
-                node = [
-                    "seq",
-                    ["assert", ["le", size, "calldatasize"]],
-                    ["mstore", pos, size],
-                    ["calldatacopy", pos + 32, 0, size],
-                    pos,
-                ]
-                return LLLnode.from_list(node, typ=typ, pos=getpos(self.expr), location="memory")
+                return LLLnode(None, typ=ByteArrayType(0), location="calldata")
```

Comment:
I think the behavior of LLLnodes with value=None is a little bit hard to reason through. Better to set it to 0 here, it will communicate better that it is just a 0-length array that starts at calldata=0.

Reply:
makes sense, I was debating between which to choose and did None, but 0 like you said communicates better